# HW 2

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [2]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [3]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

## Problem 3

Again, a sketch is a good start. It easy to calculate that in one hour at 250 mph, the plane flew 250 mi to get to point $B$, while its destination is really point $A$.

![hw2s_3](hw2s_3.png)

Since the triangle involving the origin and the two points $A$, $B$ happens to have a $90^\circ$ angle, it'd actually be possible to just trigonometry directly. But let's do it the vector way. I'm using the position vectors $\vec x_A$ and $\vec x_B$ for the two points $A$, $B$. I can see from my sketch that

$$
\vec x_B + \Delta \vec x = \vec x_A
$$

Therefore the airplane still has to fly $\Delta \vec x$ to get to its destination, where
$$
\Delta \vec x = \vec x_A - \vec x_B
$$

In order to actually calculate that, I need to split the vectors into components:
\begin{align}
\vec x_A &= 300\,\text{mi}\, (\cos 45^\circ \hat i + \sin 45^\circ \hat j)\\
\vec x_B &= 250\,\text{mi}\, (-\cos 45^\circ \hat i + \sin 45^\circ \hat j)\\
\end{align}

Python understands vectors, though it doesn't do unit vector notation, it just shows the two components of each vector:

In [18]:
x_A = np.array([300. * cos(rad(45)), 300. * sin(rad(45))])
x_B = np.array([-250. * cos(rad(45)), 250. * sin(rad(45))])
print("x_A = {}\nx_B = {}".format(x_A, x_B))

x_A = [212.13203436 212.13203436]
x_B = [-176.7766953  176.7766953]


In [19]:
delta_x = x_A - x_B
print("delta x = {}".format(delta_x))

delta x = [388.90872965  35.35533906]


So
$$
\Delta\vec x = 389 \text{ mph } \hat i + 35 \text{ mph } \hat j
$$
which still needs converting into magnitude and direction the usual way, using the Pythagorean Theorem and inverse tangent.

In [25]:
d = sqrt(delta_x[0]**2 + delta_x[1]**2)
th = deg(arctan2(delta_x[1], delta_x[0]))
print("distance = {:f}, theta = {:f}".format(d, th))

distance = 390.512484, theta = 5.194429


Finally, we need to figure out how long it takes to go 390 miles at 250 mph.

$$v = d / t \Longrightarrow t = d/v$$

In [27]:
t = d / 250
print("time = {:f} h = {:f} min".format(t, t*60))

time = 1.562050 h = 93.722996 min


So the plane has to fly for 94 minutes in the direction $5^\circ$ N of E.